zne/utils/classconstant.py

```diff
@@ -0,0 +1,49 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class constant descriptor."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Any
+
+
+class classconstant:  # pylint: disable=invalid-name
+
+    """Class constant descriptor.
+
+    Args:
+        value: the constant value to store at the class level.
+        copy: whether to return a deepcopy of the value or the original reference.
+
+    Note: returning constant values by reference is more performant but opens the up
+    the possibility to mutating them. For this reason, if the value type is not
```

Comment:
Did you mean to say "if the value type `is` immutable, ..." here?

I assume if your data type is immutable, you want this class returning copies of itself?

Reply:
Thanks for the review! 

The idea is that if the value can be mutated (i.e. the value is NOT immutable) then the descriptor should return copies instead. Otherwise the value could be updated and would not be constant 🙂